packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -224,7 +224,8 @@ public function getNativeType(Expr $expr): Type
             return new MixedType();
         }
 
-        return $scope->getNativeType($expr);
+        $type = $scope->getNativeType($expr);
+        return $this->accessoryNonEmptyStringTypeCorrector->correct($type);
```

Comment:
Basically, this reduces `string|non-empty-string` to `string`. 
The `non-empty-string` is deteted by PHPStan, but not relevant for us, as all strict type declarations are only in `string` form.